src/main/java/org/haffson/adventofcode/days/day03/Day03.java

```diff
@@ -0,0 +1,109 @@
+package org.haffson.adventofcode.days.day03;
+
+import org.haffson.adventofcode.ProblemStatusEnum;
+import org.haffson.adventofcode.days.Days;
+import org.haffson.adventofcode.utils.FileReaders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+
+/**
+ * Implementation for <i>Day 1: Chronal Calibration</i>.
+ */
+@Component
+public class Day03 implements Days {
+
+    /**
+     * The puzzle status {@code HashMap}
+     */
+    private final HashMap<String, ProblemStatusEnum> problemStatus;
+    private final FileReaders fileReaders;
+
+    /**
+     * Causes the input file to be parsed into the frequencies array ({@code frequencies}).
+     *
+     * @param fileReaders {@code @Autowired} fileReader //TODO: inject what you need
+     */
+    @Autowired
+    Day03(final FileReaders fileReaders) {
+        this.fileReaders = fileReaders;
+        this.problemStatus = new HashMap<>();
+        this.problemStatus.put("1", ProblemStatusEnum.SOLVED);
+        this.problemStatus.put("2", ProblemStatusEnum.UNSOLVED);
+    }
+
+    @Override
+    public int getDay() {
+        return 3;
+    }
+
+    @Override
+    public HashMap<String, ProblemStatusEnum> getProblemStatus() {
+        return problemStatus;
+    }
+
+    @Override
+    public String firstPart() {
+        final String fileName = "src/main/resources/puzzle_input/day3_input.txt";
+
+        return "Part 1 - two or more claims: " + calculateClaims(fileReaders.getInputList(fileName));
+    }
+
+    @Override
+    public String secondPart() {
+        return "Part 2: " + null;
+    }
+
+    /**
+     * Primary method for Day 3, Part 1.
+     * Calculates the final frequency as the sum of all frequencies.
+     *
+     * @return the final frequency
+     */
+
+
+    private long calculateClaims(final List<String> myArrayList) {
+        final List<RectangleClaim> rectangleClaimList = convertStringToRectangleList(myArrayList);
+
+        final String[][] fabric = new String[1000][1000];
+        rectangleClaimList.forEach(claim ->
+                IntStream.range(claim.getLeftSpace(), claim.getLeftSpace() + claim.getWidth())
```

Comment:
Ausdrücke wie `claim.getLeftSpace() + claim.getWidth()` kann man auch immer gut in Hilfsmethoden (in diesem Fall auf `RectangleClaim`) überführen: Das dürfte die rechte X-Koordinate des Claims sein.

Reply:
done